src/main/java/core/basesyntax/strategy/BalanceHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class BalanceHandler implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Map<String, Integer> inventory) {
+        inventory.put(transaction.getFruit(),
```

Comment:
The 'executeOperation' method is using 'Math.max' which is not the correct logic for maintaining the balance in the inventory. The balance should represent the actual quantity in inventory, not the maximum of two quantities.